lib/tracker.coffee

```diff
@@ -0,0 +1,66 @@
+# contants
+analyticsWriteKey = 'pDV1EgxAbco4gjPXpJzuOeDyYgtkrmmG'
+
+# imports
+_ = require 'underscore-plus'
+{allowUnsafeEval} = require 'loophole'
+
+# Analytics require a special import because of [Unsafe-Eval error](https://github.com/Glavin001/atom-beautify/commit/fbc58a648d3ccd845548d556f3dd1e046075bf04)
+Analytics = null
+allowUnsafeEval -> Analytics = require 'analytics-node'
+
+# load package.json to include package info in analytics
+pkg = require("../package.json")
+
+class Tracker
+
+  constructor: (@analyticsUserIdConfigKey) ->
+    # Setup Analytics
+    @analytics = new Analytics analyticsWriteKey
+
+    # set a unique identifier
+    if not atom.config.get @analyticsUserIdConfigKey
+      uuid = require 'node-uuid'
+      atom.config.set @analyticsUserIdConfigKey, uuid.v4()
+
+    # identify the user
+    atom.config.observe @analyticsUserIdConfigKey, {}, (userId) =>
```

Comment:
The empty options `{}` could be omitted; https://atom.io/docs/api/v1.0.2/Config#instance-observe